core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -65,6 +65,21 @@ struct identity {
 template <typename T>
 using identity_t = typename identity<T>::type;
 
+struct not_a_type {
+  not_a_type()                  = delete;
+  ~not_a_type()                 = delete;
+  not_a_type(not_a_type const&) = delete;
+  void operator=(not_a_type const&) = delete;
+};
```

Comment:
Non-blocking and super minor: In detection idiom, this is usually named `nonesuch`. By convention in metaprogramming (for whatever reason), `not_a_type` (or `nat` sometimes) is usually just an undefined type or completely empty at the least.

Reply:
We can rename when we add the detection idiom